<a href="https://colab.research.google.com/github/Carla9711/Machine-Learning---Classification/blob/main/Machine_Learning_Classification.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Machine Learning - Classification

In [1]:
# Mount google drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
# import libraries
import pandas as pd

# Stroke Prediction Dataset

## Data Dictionary

* id: unique identifier
* gender: "Male", "Female" or "Other"
* age: age of the patient
* hypertension: 0 if the patient doesn't have hypertension, 1 if the patient has hypertension
* heart_disease: 0 if the patient doesn't have any heart diseases, 1 if the patient has a heart disease
* ever_married: "No" or "Yes"
* work_type: "children", "Govt_jov", "Never_worked", "Private" or "Self-employed"
* Residence_type: "Rural" or "Urban"
* avg_glucose_level: average glucose level in blood
* bmi: body mass index
* smoking_status: "formerly smoked", "never smoked", "smokes" or "Unknown"*
* stroke: 1 if the patient had a stroke or 0 if not

## Load Data

In [24]:
path2 = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week07/Data/healthcare-dataset-stroke-data.csv'

In [25]:
df_stroke = pd.read_csv(path2)

In [26]:
df_stroke.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [27]:
df_stroke.shape

(5110, 12)

## Dataset 2 Q&A

1. [Source of data](https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset)

2. Brief description of data
  * Health and demograohic data collected by WHO

3. What is the target?
  * Whether a patient is likely to have a stroke or not

4. What does one row represent? (A person? A business? An event? A product?)
  * One row represents a person

5. Is this a classification or regression problem?
  * Classification

6. How many features does the data have?
  * 11

7. How many rows are in the dataset?
  * 5110

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
 * None so far

## Clean Data

* Delete unnecessary columns.
* Delete duplicate rows.
* Identify and addressed missing values.
* Identify and correct inconsistencies in data for categorical values (i.e. Cat, cat, cats).
* Produce univariate visuals for the target and all features.
* Identify outliers.


In [28]:
df_stroke.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [29]:
df_stroke['id'].nunique()

5110

* Each row represents a person with a unique ID
* ID will be dropped as it's not a useful machine learning feature

In [30]:
# drop unnecessary columns
df_stroke.drop(columns=['id'], inplace=True)

In [32]:
# sum number of duplicate rowa
df_stroke.duplicated().sum()

0

* There are no duplicate rows to delete

In [ ]:
# check value counts of categorical data


In [33]:
# identify missing values
df_stroke.isna().sum()

gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

* There are 201 missing bmi values

# Adult income dataset

[Source: Data Dictionary](https://www.kaggle.com/code/jieyima/income-classification-model)

## Data Dictionary

* age
  * Age of an individual

* workclass: Private, Self-emp-not-inc, Self-emp-inc, Federal-gov, Local-gov, State-gov, Without-pay, Never-worked.
  * Individual work category

* fnlwgt: final weight
  * The weights on the CPS files are controlled to independent estimates of the civilian noninstitutional population of the US. These are prepared monthly by the Population Division at the Census Bureau.

* education: Bachelors, Some-college, 11th, HS-grad, Prof-school, Assoc-acdm, Assoc-voc, 9th, 7th-8th, 12th, Masters, 1st-4th, 10th, Doctorate, 5th-6th, Preschool.
  * Individual's highest education degree

* education-num: number of education year, continuous.
  * Individual's year of receiving education

* marital-status: Married-civ-spouse, Divorced, Never-married, Separated, Widowed, Married-spouse-absent, Married-AF-spouse.
 * Individual marital status

* occupation: Tech-support, Craft-repair, Other-service, Sales, Exec-managerial, Prof-specialty, Handlers-cleaners, Machine-op-inspct, Adm-clerical, Farming-fishing, Transport-moving, Priv-house-serv, Protective-serv, Armed-Forces.
  * Individual's occupation

* relationship: Wife, Own-child, Husband, Not-in-family, Other-relative, Unmarried.
  * Individual's relation in a family

* race: White, Asian-Pac-Islander, Amer-Indian-Eskimo, Other, Black.
  * Race of Individual

* sex: Female, Male.

* capital-gain
* capital-loss
* hours-per-week:
  * Individual's working hour per week

* native-country: United-States, Cambodia, England, Puerto-Rico, Canada, Germany, Outlying-US(Guam-USVI-etc), India, Japan, Greece, South, China, Cuba, Iran, Honduras, Philippines, Italy, Poland, Jamaica, Vietnam, Mexico, Portugal, Ireland, France, Dominican-Republic, Laos, Ecuador, Taiwan, Haiti, Columbia, Hungary, Guatemala, Nicaragua, Scotland, Thailand, Yugoslavia, El-Salvador, Trinadad&Tobago, Peru, Hong, Holand-Netherlands.
  * Individual's native country








In [3]:
path1 = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week07/Data/adult.csv'
df_adult = pd.read_csv(path1)

In [4]:
df_adult.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [5]:
df_adult.shape

(48842, 15)

## Dataset 1 Q&A

1. [Source of data](https://www.kaggle.com/datasets/wenruliu/adult-income-dataset)

2. Brief description of data
  * Demographic and income data colected in a census

3. What is the target?
  * An individuals income

4. What does one row represent? (A person? A business? An event? A product?)
  * One row represents a person

5. Is this a classification or regression problem?
  * Classification

6. How many features does the data have?
  * 14

7. How many rows are in the dataset?
  * 48842

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
  * Some columns such as fnlwgt (final weight) might be hard to intepret as it's an attribute specific to the United States